core/src/main/java/com/klaytn/caver/abi/ABI.java

```diff
@@ -0,0 +1,317 @@
+package com.klaytn.caver.abi;
+
+import com.klaytn.caver.contract.ContractEvent;
+import com.klaytn.caver.contract.ContractIOType;
+import com.klaytn.caver.contract.ContractMethod;
+import org.web3j.abi.*;
+import org.web3j.abi.datatypes.*;
+import org.web3j.crypto.Hash;
+import org.web3j.utils.Numeric;
+
+import java.lang.reflect.InvocationTargetException;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Representing a ABI type encode / decode.
+ */
+public class ABI {
+
+    /**
+     * Encodes a function call.
+     * @param method A ContractMethod instance.
+     * @param params A List of method parameter.
+     * @return String
+     */
+    public static String encodeFunctionCall(ContractMethod method, List<Object> params) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+        String functionSignature = ABI.buildFunctionSignature(method);
+
+        List<String> solTypeList = new ArrayList();
+
+        for (ContractIOType contractIOType : method.getInputs()) {
+            solTypeList.add(contractIOType.getType());
+        }
+
+        return encodeFunctionCall(functionSignature, solTypeList, params);
+    }
+
+    /**
+     * Encodes a function call.
+     * @param functionSig A function signature string.
+     * @param params A List of method parameter.
+     * @return String
+     */
+    public static String encodeFunctionCall(String functionSig, List<String> solTypeList, List<Object> params) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+        String methodId = encodeFunctionSignature(functionSig);
+        String encodedParams = encodeParameters(solTypeList, params);
+
+        return methodId + encodedParams;
+    }
+
+    /**
+     * Encodes a function signature.
+     * @param method A ContractMethod instance.
+     * @return String
+     */
+    public static String encodeFunctionSignature(ContractMethod method) {
+        return encodeFunctionSignature(buildFunctionSignature(method));
+    }
+
+    /**
+     * Encodes a function signature.
+     * @param functionSig A function signature string.
+     * @return String
+     */
+    public static String encodeFunctionSignature(String functionSig) {
+        byte[] input = functionSig.getBytes();
+        byte[] hash = Hash.sha3(input);
+        return Numeric.toHexString(hash).substring(0, 10);
+    }
+
+    /**
+     * Build a function signature.
+     * @param method A ContractMethod instance.
+     * @return String
+     */
+    public static String buildFunctionSignature(ContractMethod method) {
```

Comment:
How about changing this function name to `buildFunctionString` or something else?

Reply:
You are right. I will change to buildFunctionString